### PR TITLE
Add experiment tracking using mlflow

### LINKED ISSUE #232: Tracking of flownet runs
To be able to find the best configuration for the Norne model we will have to test a lot of hyper parameters. Some aid with keeping track of parameters and model results is needed. We could consider using [mlflow](https://www.mlflow.org/) for this task.

### PR DESCRIPTION
**Pull request by wouterjdb**
This PR adds the possibility to run multiple flownet configuration files and do experiment tracking using mlflow.

---

### Contributor checklist

- [x] :tada: This PR closes #232, #233 
- [x] :scroll: I have broken down my PR into the following tasks:
   - [x] Add mlflow dependency
   - [x] Add hyperopt dependency
     - [x] Expose choice between random search  and optimization
   - [x] Add hyperparamter running mode in arg parse
   - [x] Parse configuration file with ranges and choices
     - [x] Initial parsing
     - [x] Add flexibility for user
   - [x] Generate mlflow experiment
     - [x] Run flownet
     - [x] Generate metrics and store in mlflow
- [x] :robot: I have added tests, or extended existing tests, to cover any new features or bugs fixed in this PR.
- [x] :book: I have considered adding a new entry in `CHANGELOG.md`.
- [x] :books: I have considered updating the documentation.

### CODE REVIEW

**Comment by wouterjdb** [comment]
Tested now with the Egg CI Model. The workflow is working using hyperopt and visualization/tracking with MLFlow:

image

I track however a metric of "1" at the moment. Still a todo. Let's look at this tomorrow @edubarrosTNO.

**Comment by wouterjdb** [comment]
Added metrics and parameters. Exposed hyperopt options in config. image

File: .github/workflows/flownet.yml [hunk 1]
```diff
@@ -46,10 +46,10 @@ jobs:
           INPUT_MODEL_FOLDER: ./flownet-testdata/${{ matrix.flownet-model }}/input_model
           # If you want the CI to (temporarily) run against your fork of the testdada,
           # change the value her from "equinor" to your username.
-          TESTDATA_REPO_OWNER: equinor
+          TESTDATA_REPO_OWNER: wouterjdb
           # If you want the CI to (temporarily) run against another branch than master,
           # change the value her from "master" to the relevant branch name.
-          TESTDATA_REPO_BRANCH: master
+          TESTDATA_REPO_BRANCH: add_hyperparam_config
```
Comment (olelod) [inline .github/workflows/flownet.yml:52]: Remember to change back before merging

**Review by olelod — APPROVED** [review]
Nice work. LGTM.

**Pull request merged**